core/http-auth/src/main/java/software/amazon/awssdk/http/auth/NoAuthAuthScheme.java

```diff
@@ -34,6 +34,14 @@ static NoAuthAuthScheme create() {
         return DefaultNoAuthAuthScheme.create();
     }
 
+    /**
+     * Retrieve the scheme ID.
+     */
+    @Override
+    default String schemeId() {
+        return SCHEME_ID;
+    }
```

Comment:
note the default implementation of `schemeId()` is kept in the interface as opposed to the Default*AuthScheme impl. This is the case in the other schemes already, and changed it in NoAuthAuthScheme to match others.

Q: Should there be no default implementation of `schemeId()` in the interface too?

Reply:
I don't see any benefit of either approach over the other, so I suppose for consistency sake, just move it to the default impl and get rid of default here and in the other auth-scheme interfaces

Reply:
I guess if `SCHEME_ID`is referenced outside of the module, it makes sense to be defined in this interface.

Reply:
SCHEME_ID will remain defined in the interface.
Based on offline discussions, will remove the default implementation from the interface.

Reply:
Removed in https://github.com/aws/aws-sdk-java-v2/pull/4416/commits/a916ca06af5f3d97b4931a57a8a070fc7e619409